pyresample/geometry.py

```diff
@@ -547,6 +547,39 @@ def __init__(self, lons, lats, nprocs=1):
         elif lons.ndim > 2:
             raise ValueError('Only 1 and 2 dimensional swaths are allowed')
 
+    def copy(self):
+        return SwathDefinition(self.lons, self.lats)
```

Comment:
Shouldn't there be a check and special handling of numpy arrays?

Reply:
That should be done in the constructor then

Reply:
And it is